### PR TITLE
DOC: stats: Convert `chisquare` example to notebook

### PR DESCRIPTION
**Pull request by melissawm**
#### Reference issue
--

#### What does this implement/fix?
This PR moves the narrative example from the chisquare hypothesis test docstring to a markdown-based notebook.

cc @mdhaber

### CODE REVIEW

File: doc/source/tutorial/stats/hypothesis_chisquare.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  name: python3
+---
+
+```{eval-rst}
+.. jupyterlite:: ../../_contents/hypothesis_chisquare.ipynb
+   :new_tab: False
+```
+
+(hypothesis_chisquare)=
+# Hypothesis testing: Chi-square test
```
Comment (mdhaber) [inline doc/source/tutorial/stats/hypothesis_chisquare.md:20]: If the other examples going here follow the precedent, all will begin "Hypothesis testing". Is that intentional?

File: doc/source/tutorial/stats/hypothesis_tests.rst [hunk 2]
```diff
@@ -10,5 +10,12 @@ Sample statistics and hypothesis tests
 
 Statistical hypothesis tests are used to decide whether data sufficiently
 support a particular hypothesis. SciPy defines a number of hypothesis tests,
-listed in :ref:`hypotests`. You can find examples to each test in the
-corresponding docstring.
+listed in :ref:`hypotests`.
+
+You can find simple examples to each test in the corresponding docstring. For
```
Comment (mdhaber) [inline doc/source/tutorial/stats/hypothesis_tests.rst:15]:
```suggestion
listed in :ref:`hypotests`. You can find simple examples to each test in the corresponding docstring. For
```
Also, I think I'll have some suggestions for the layout of this page when we have some more examples moved over, but it is OK for now.

File: scipy/stats/_stats_py.py [hunk 3]
```diff
@@ -7742,52 +7743,15 @@ def chisquare(f_obs, f_exp=None, ddof=0, axis=0):
 
     Examples
     --------
```
Comment (mdhaber) [inline scipy/stats/_stats_py.py:7745]: I see the refernence in "See also" above, and that looks like a good idea. Can we also add a link at the end of the example? E.g., "For a more detailed example, see..."?

**Review by mdhaber — COMMENTED** [review]
Checked the rendered output and compared the diff - looks good! Besides addressing the CI failure due to missing imports, just a few things.

File: doc/source/tutorial/stats/hypothesis_tests.rst (hunk 2 above)
Comment (melissawm) [reply doc/source/tutorial/stats/hypothesis_tests.rst:15]: Yeah, this is pretty much a placeholder for now, happy to restructure.

File: scipy/stats/_stats_py.py (hunk 3 above)
Comment (melissawm) [reply scipy/stats/_stats_py.py:7745]: Will do 👍🏻

File: doc/source/tutorial/stats/hypothesis_chisquare.md (hunk 1 above)
Comment (melissawm) [reply doc/source/tutorial/stats/hypothesis_chisquare.md:20]: I was of two minds about it so we can remove if we feel this is not needed, especially if these documents all live under a toc with "Hypothesis testing" caption/title or something similar.

**Comment by mdhaber** [comment]
Great. I'll merge when I see those changes and that CI is green. Thank you!

File: doc/source/tutorial/stats/hypothesis_chisquare.md (hunk 1 above)
Comment (mdhaber) [reply doc/source/tutorial/stats/hypothesis_chisquare.md:20]: I think they will.

**Comment by melissawm** [comment]
If you are happy with how this looks, I'll be sending the others soon. I am guessing one PR per example sounds more reasonable, so I may send in several at a time but please don't feel pressured to review them all quickly, take your time 😄 Thanks!

**Comment by mdhaber** [comment]
> I am guessing one PR per example sounds more reasonable,

Normally, that would be my preference, but actually, these seem straightworward enough to include all in one go if you prefer.

**Comment by melissawm** [comment]
Great. I'll pull the ones I have together and submit soon.

**Comment by mdhaber** [comment]
Great. I'll go ahead and merge this one. Two other little things I noticed to keep in mind.

https://output.circle-artifacts.com/output/job/2262f2c8-1c2f-4b7f-b157-f7c24f76fb58/artifacts/0/html/tutorial/stats.html shows 

image

I think that if the function-specific examples are shown in that TOC, they should be at a lower level (more indentation).

Also, in

image

Both before and now, it's not immediately obvious what the link is for. Is there a way for it to render here like "Extended example: Chi-square test" here even if the page title is just "Chi-Square Test"?

**Comment by melissawm** [comment]
Makes sense. I can fix those in the follow-up, if that's ok with you.

**Pull request merged**